testutil/integration/evmos/factory/build.go

```diff
@@ -76,3 +77,81 @@ func (tf *IntegrationTxFactory) GenerateSignedEthTx(privKey cryptotypes.PrivKey,
 
 	return tf.buildSignedTx(signedMsg)
 }
+
+// GenerateMsgEthereumTx creates a new MsgEthereumTx with the provided arguments.
+// If any of the arguments are not provided, they will be populated with default values.
+func (tf *IntegrationTxFactory) GenerateMsgEthereumTx(
+	privKey cryptotypes.PrivKey,
+	txArgs evmtypes.EvmTxArgs,
+) (evmtypes.MsgEthereumTx, error) {
+	fromAddr := common.BytesToAddress(privKey.PubKey().Address().Bytes())
+	// Fill TxArgs with default values
+	txArgs, err := tf.populateEvmTxArgs(fromAddr, txArgs)
```

Comment:
Maybe we can rename the func: `populateEvmTxArgsWithDefault` so we don't have to add explanatory comments because the signature is more descriptive. wdyt?